### PR TITLE
[Feat] - Updated Sessions.js

### PR DESCRIPTION
**Pull request by adithyaakrishna**
**Description:**

- Used Array.from() instead of .push()

### CODE REVIEW

**Comment by adithyaakrishna** [comment]
@dkoes Could you pls lmk if these changes looks good?

File: learning-environment/sessions.js [hunk 1]
```diff
@@ -298,10 +298,7 @@ var initSessions = function() {
     //return information about what is currently clicked for the server
     var getClicked = function() {
         var ret = [];
```
Comment (dkoes) [inline learning-environment/sessions.js:300]: need to remove the var ret delaration

**Review by dkoes — CHANGES_REQUESTED** [review]

File: learning-environment/sessions.js [hunk 2]
```diff
@@ -386,22 +383,20 @@ var initSessions = function() {
...
-            for (let i = 0; i < query_result.length; i++) {
-                let pos = query_result[i][0];
-                let p = {x: pos[0], y: pos[1], z: pos[2]};
-                let val = parseInt(query_result[i][1]);                
-                let l = glviewer.addLabel(val, {position : p,backgroundOpacity:0.5+(val/max)*.5});                
-                result_labels.push(l);                
-            }
+        const result_labels = Array.from(query_result, ([pos, val]) => {
+            const p = { x: pos[0], y: pos[1], z: pos[2] };
+            const label = glviewer.addLabel(val, { position: p, backgroundOpacity: 0.5 + (val / max) * 0.5 });
```
[the remark this answers is not in the document]
Comment (adithyaakrishna) [reply learning-environment/sessions.js]: Oops! My bad, thanks for pointing it out :)

File: learning-environment/sessions.js [hunk 3]
```diff
@@ -386,22 +382,28 @@ var initSessions = function() {
...
-    });
+        
+        // First clear existing labels
+        clearResultLabels();
+        
+        let max = 1;
+        let i = 0;
+      
+        // Calculate max
+        while (i < query_result.length) {
```
Comment (dkoes) [inline learning-environment/sessions.js:393]: why was this changed to a while loop? Using a while loop for simple iteration is poor coding practice. If you want a more modern pattern use forEach.

**Review by dkoes — CHANGES_REQUESTED** [review]

File: learning-environment/sessions.js (hunk 3 above)
Comment (adithyaakrishna) [reply learning-environment/sessions.js:393]: @dkoes I tried benching them now, their performance benefit compared to forEach is a bit on the lower end, I will make the change and send it in sometime :)

**Review by dkoes — APPROVED** [review]
